fairlearn/metrics/_metric_frame.py

```diff
@@ -0,0 +1,515 @@
+        functions return scalar values). If there are no control
+        features, then the result will be a :class:`pandas.Series`,
+        with rows corresponding to the underying metrics.
+
+        Returns
+        -------
+        pandas.Series or pandas.DataFrame
+            If there are control features, then the result is a
+            :class:`pandas.DataFrame` with columns corresponding to the
+            underlying metrics, and rows indexed by the classes identified
+            by the control features. Otherwise, a :class:`pandas.Series`
+            indexed by the names of the underlying metrics.
+        """
+        if not self.control_levels:
+            result = pd.Series(index=self.by_group.columns, dtype='object')
+            for m in result.index:
+                min_val = self.by_group[m].min()
+                result[m] = min_val
+        else:
+            result = self.by_group.groupby(level=self.control_levels).min()
+        return result
+
+    def difference(self,
+                   method: str) -> Union[pd.Series, pd.DataFrame]:
+        """Return the maximum absolute difference between groups for each metric.
+
+        This method calculates a scalar value for each underlying metric by
+        finding the maximum absolute difference between the entries in each
+        column of the ``by_groups`` DataFrame.
+
+        Similar to other methods, the result will be a :class:`pandas.Series`
+        if there are no control features, with rows corresponding
+        to each underlying metric. If there are control features,
+        then the result will be a :class:`pandas.DataFrame` with a column
+        for each underlying metric, and rows corresponding to the subgroups
+        identified by the control feature(s).
+
+        There are two allowed values for the ``method=`` parameter. The
+        value ``between_groups`` computes the maximum difference between
+        any two pairs of groups in the ``by_groups`` DataFrame (i.e.
+        ``group_max() - group_min()``). Alternatively, ``to_overall``
+        computes the difference between each subgroup and the
+        corresponding value from ``overall`` (if there are control
+        features, then ``overall`` is multivalued for each metric).
+        The result is the absolute maximum of these values.
+
+        Returns
+        -------
+        pandas.Series or pandas.DataFrame
+            If there are control features, then the result is a
+            :class:`pandas.DataFrame` with columns corresponding to the
+            underlying metrics, and rows indexed by the classes identified
+            by the control features. Otherwise, a :class:`pandas.Series`
+            indexed by the names of the underlying metrics.
+        """
+        subtrahend = np.nan
+        if method == 'between_groups':
+            subtrahend = self.group_min()
+        elif method == 'to_overall':
+            subtrahend = self.overall
```

Comment:
simplify to `return (self.by_group - self.overall).abs().max(level=control_levels)`?

Reply:
Have done some simplification.